lib/CorresK_Method.thy

```diff
@@ -611,9 +609,18 @@ private method corresK_apply =
 
 private method corresK_alternate = corresK_inst | corresK_rv
 
-
+(* declared [corres] in Corres_Method and safe there, but not here, because they are not strictly
+   terminal rules *)
+named_theorems corres_unsafeK
+lemmas [corres_unsafeK] =
```

Comment:
I thought these changes (the ones to `CorresK_Method`) made sense when I first looked at this diff, but now I'm completely confused and the comment above seems cryptic.

Reply:
the two methods share the `[corres]` set, and my method is less susceptible to unsafe rules being in the `[corres]` set, because it has all the safeguards. And because that works, I'm promptly abusing it, which then breaks `corresK`. This `corres_unsafeK` set are those unsafe rules that `corresK` should remove before it does stuff.

This idea is that normal users will follow the instructions and not put new unsafe rules in, so it should never be an issue later on, just for initial setup. (If users do put in unsafe rules, the can still add them to this set to preserve `corresK`).

Reply:
Added more of that to the comment in the file.